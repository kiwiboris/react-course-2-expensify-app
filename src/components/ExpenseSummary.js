import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectTotalExpenses from '../selectors/expenses-total';

export const ExpenseSummary = (props) => (
    <div>
        { props.expenseCount === 0 ? (
            <p></p>
        ) : (
            <p>Viewing {props.expenseCount} expenses totalling {numeral(props.expensesTotal / 100).format('$0,0.00')}</p>
        )
    }
    </div>
);

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
       expensesTotal: selectTotalExpenses(visibleExpenses),
       expenseCount: visibleExpenses.length
      };
};

export default connect(mapStateToProps)(ExpenseSummary);