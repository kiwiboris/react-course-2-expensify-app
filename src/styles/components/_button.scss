.button {
    // background-color: rgb(43, 145, 170);
    background-color: $blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size $s-size;
    text-decoration: none;
}

.button--link {
    background: none;
  }

.button--secondary {
    background: #888888;
  }