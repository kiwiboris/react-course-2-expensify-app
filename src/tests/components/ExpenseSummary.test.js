import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('should render ExpenseSummary with empty message', () => {
    const wrapper = shallow(<ExpenseSummary />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with all expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={2} expensesTotal={200} />);
    expect(wrapper).toMatchSnapshot();
});