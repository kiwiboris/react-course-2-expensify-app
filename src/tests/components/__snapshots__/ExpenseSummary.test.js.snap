// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with all expenses 1`] = `
<div>
  <p>
    Viewing 
    2
     expenses totalling 
    $2.00
  </p>
</div>
`;

exports[`should render ExpenseSummary with empty message 1`] = `
<div>
  <p>
    Viewing 
     expenses totalling 
    $0.00
  </p>
</div>
`;
